File: backend/SpareHub/Domain/MySql/VesselAtPort.cs
namespace Domain;

public class VesselAtPort
{
    public int VesselId { get; init; }
    public int PortId { get; init; }
    public DateTime? ArrivalDate { get; set; }
    public DateTime? DepartureDate { get; set; }

    public Vessel Vessel { get; set; } = null!;
    public Port Port { get; set; } = null!;
}

File: frontend/src/interfaces/vesselAtPort.ts
import { Vessel } from "./order";

export interface VesselAtPort {
    portName: string;
    portId: string;
    vessels: Vessel[];
    arrivalDate: string;
    departureDate?: string;
}


export interface VesselAtPortDetail extends VesselAtPort {
    vessels: Vessel[];
    portId: string;
    arrivalDate: string;
    departureDate?: string;
}


export interface VesselAtPortRequest {
    vesselId: string;
    portId: string;
    arrivalDate: string;
    departureDate?: string;
} 